### PR TITLE
ci: enable Vagrant-based testing using GitHub Actions

### PR DESCRIPTION
**Pull request by myii**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [x] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

* Semi-automated using https://github.com/myii/ssf-formula/pull/304

### Describe the changes you're proposing

Explaining commit-by-commit:

#### ci: enable Vagrant-based testing using GitHub Actions

* As explained at the URL mentioned above.

#### docs(readme): add `Testing with Vagrant` section

* Explain how to run these Vagrant-based tests locally.

#### fix(osfamilymap): use `whoami` for Windows `rootuser`

* Fix bug #495.

#### test: add Windows map verification files

* Add `map.jinja` verification files for all Windows platforms.

#### test(_mapdata): add Arch Linux verification file

* Add `map.jinja` verification files for Arch Linux.

#### refactor(retry_options): provide options from defaults/pillar

* Centralise `retry_options` as done in many other formulas.

#### fix(freebsd): update to provide support for most recent versions

* Add the new data in `osmap.yaml`.
* Make adjustments to ensure all instances pass in the CI.
* Add `map.jinja` verification files for FreeBSD.

#### fix(openbsd): update to provide support for most recent versions

* Add the new data in `osmap.yaml`.
* Add `map.jinja` verification files for OpenBSD.

#### test: update for new platforms (inc. using `system` library)

* Update all tests to include the new platforms.
* Use the `system` library throughout.

### Pillar / config required to test the proposed changes

As-is.

### Debug log showing how the proposed changes work

As shown in the CI.

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [x] Included in Kitchen (i.e. under `state_top`).
- [x] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### CODE REVIEW

File: test/integration/v3000-py2/controls/service_spec.rb [hunk 1]
```diff
@@ -1,12 +1,19 @@
 # frozen_string_literal: true
 
+services =
+  case platform[:family]
```
Comment (baby-gnu) [inline test/integration/v3000-py2/controls/service_spec.rb:4]: Should it be `system.platform[:family]` if at some point in the future we need to do special stuffs for `arch` or `gentoo` and be consistent with other places?

File: test/integration/v3000-py3/controls/service_spec.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 services =
   case platform[:family]
```
Comment (baby-gnu) [inline test/integration/v3000-py3/controls/service_spec.rb:4]: Same as above, shouldn't we switch to `system.platform[:family]`?

File: test/integration/v3002-py3/controls/service_spec.rb [hunk 3]
```diff
@@ -1,12 +1,19 @@
 # frozen_string_literal: true
 
+services =
+  case platform[:family]
```
Comment (baby-gnu) [inline test/integration/v3002-py3/controls/service_spec.rb:4]: Same as above, shouldn't we switch to `system.platform[:family]`?

File: test/integration/v3001-py3/controls/service_spec.rb [hunk 4]
```diff
@@ -1,12 +1,19 @@
 # frozen_string_literal: true
 
+services =
+  case platform[:family]
```
Comment (baby-gnu) [inline test/integration/v3001-py3/controls/service_spec.rb:4]: Same as above, shouldn't we switch to `system.platform[:family]`?

**Review by baby-gnu — COMMENTED** [review]
Maybe homogenize the use of `system.platform` over `platform`?

File: test/integration/v3000-py2/controls/service_spec.rb (hunk 1 above)
Comment (myii) [reply test/integration/v3000-py2/controls/service_spec.rb]: @baby-gnu I suppose my thought pattern had been to use pure InSpec as much as possible and only rely on our library when necessary.  Anyway, I've taken your suggestion and applied it -- thanks for that.

**Review by baby-gnu — APPROVED** [review]

**Pull request merged**